### PR TITLE
Bootstrap

### PR DESCRIPTION
**Pull request by crocodoyle**
This pull request completely refactors the CBRAIN User Interface using Twitter Bootstrap.

### CODE REVIEW

**Comment by natacha-beck** [comment]
@crocodoyle here is some comments about visual bug I saw: 

- In `home` when you resize the window (for example to the half of my screen) the list under the `Tools available to you` will go out of there div. Same for the `Latest Uptated Tasks . `
- On `portal_log`: 
  - Seems to have issue you have `Settings Filter Filter` it is strange to read. 
  - When you resize the windows the left part with the Filter start to overlap. 
- On `custom_filters/new`, the `Filtering by date` div do not match the main div when resizing. On a more generic note for this page, I think we can use more space for the form, and avoid to have this one stuck only on half of the page. 
- The page for the task custom filter is not accessible.

**Comment by natacha-beck** [comment]
Some more comments: 
- On the `File Upload` panel, the `Browse` button is not well align. And the `Allow modification by other project members` start to overlap the checkbox when resizing. 
- If you used the `Search` overall and start to resize the page you will have strange layout once again.
- On the `userfiles` page why the `Extract Files form Collection` is on the right of the list, and not down of the list ? 
- On `sites/new` you have some issue when resizing, the `Active Users` rows go out of the div.

**Comment by crocodoyle** [comment]
@natacha-beck I'm not sure what you mean by:
-Strange layout in the `Disk Usage report` and `Server Workdir Size Report`. 
-row of the Mail configiration and of Cache Management... miss some lines

File: BrainPortal/app/controllers/custom_filters_controller.rb [hunk 1]
```diff
@@ -58,19 +62,34 @@ def create #:nodoc:
...
+    end
+
+    respond_to do |format|
+      format.html { redirect_to :controller => :userfiles, :action => :index, :format => :html } if filter_param == UserfileCustomFilter.name
+      format.html { redirect_to :controller => :tasks, :action => :index, :format => :html } if filter_param == TaskCustomFilter.name
     end
 
   end
 
+  def show #:nodoc:
```
Comment (prioux) [inline BrainPortal/app/controllers/custom_filters_controller.rb]: the show custom filter page layout should be like the new custom filter page layout; they look different

File: BrainPortal/app/controllers/custom_filters_controller.rb [hunk 2]
```diff
@@ -82,14 +101,14 @@ def update #:nodoc:
...
     if @custom_filter.errors.empty?
       flash[:notice] = "Custom filter '#{@custom_filter.name}' was successfully updated."
-      return
     end
 
-
     respond_to do |format|
-      format.xml  { render :xml => @custom_filter.errors, :status => :unprocessable_entity }
-      format.js
+      format.html { redirect_to :controller => :userfiles, :action => :index, :format => :html } if filter_param == UserfileCustomFilter.name
```
Comment (prioux) [inline BrainPortal/app/controllers/custom_filters_controller.rb]: update  shoudl render edit page again if there are errors

File: BrainPortal/app/helpers/documentation_helper.rb [hunk 3]
```diff
@@ -32,11 +32,12 @@ module DocumentationHelper
   # shown documentation page.
   # See the help_document views for more details.
   def help_button(key, display = "Help")
-    doc = HelpDocument.find_by_key(key) || HelpDocument.from_existing_file!(key)
+    doc = HelpDocument.find_by_key(key)
```
Comment (prioux) [inline BrainPortal/app/helpers/documentation_helper.rb]: I don't know why the `HelDocument.from_existing_file!(key)` disappeared, and what did it do.

File: BrainPortal/app/views/custom_filters/_custom_filter_list.html.erb [hunk 4]
```diff
@@ -18,20 +18,68 @@
...
 # You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.  
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 -%>
 
-<h4>By Custom Filter</h4>
 
-<%= overlay_ajax_link 'Create Custom Filter', {:controller => :custom_filters, :action => :new, :filter_class => @custom_filter.class.name} %>
+<li>
```
Comment (crocodoyle) [inline BrainPortal/app/views/custom_filters/_custom_filter_list.html.erb]: The <ul>'s have attributes that are specific to the page, so they're not in the partial.

File: BrainPortal/app/helpers/show_table_helper.rb [hunk 5]
```diff
@@ -347,13 +349,13 @@ def show_table(object, options = {})
...
-      html << "<BR>"
       if object.new_record?
-        html << submit_button("Create")
+        html << submit_button("Create", { :class => "btn-block"})
       else
-        html << submit_button("Update")
+        html << submit_button("Update", { :class => "btn-block"})
       end
       html << "</div>"
+      html << "</div>"
```
Comment (prioux) [inline BrainPortal/app/helpers/show_table_helper.rb]: You are missing a closing div if the table is not editable

**Pull request merged**